ocaml/message-switch/core/make.ml

```diff
@@ -352,26 +352,36 @@ functor
 
     let listen ~process ~switch:port ~queue:name () =
       let token = Printf.sprintf "%d" (Unix.getpid ()) in
-      M.connect port >>= fun c ->
-      Connection.rpc c (In.Login token) >>|= fun (_ : string) ->
-      Connection.rpc c (In.CreatePersistent name) >>|= fun _ ->
+      let reconnect () =
+        M.connect port >>= fun request_conn ->
+        Connection.rpc request_conn (In.Login token) >>|= fun (_ : string) ->
+        Connection.rpc request_conn (In.CreatePersistent name) >>|= fun _ ->
+        M.connect port >>= fun reply_conn ->
```

Comment:
copy/paste error? why are you logging in twice?

Reply:
for two different connections? I just followed what `protocol_unix` did

```
 do_rpc request_conn (In.Login token) >>|= fun (_ : string) ->
      protect_connect switch @@ fun reply_conn ->
      do_rpc reply_conn (In.Login token) >>|= fun (_ : string) ->
```

Reply:
Ah nevermind the copy/paste error is in Protocol_unix, this one is fine.
Anyway if we delete most of protocol_unix and rely on the code here that'll remove the duplication